api/v1/contributorRoles/ContributorRoleController.php

```diff
@@ -0,0 +1,291 @@
+<?php
+
+/**
+ * @file api/v1/contributorRoles/ContributorRoleController.php
+ *
+ * Copyright (c) 2025 Simon Fraser University
+ * Copyright (c) 2025 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class ContributorRoleController
+ *
+ * @brief Handle API requests for contributor role operations.
+ *
+ */
+
+namespace PKP\API\v1\contributorRoles;
+
+use APP\core\Application;
+use APP\facades\Repo;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\Route;
+use PKP\core\PKPBaseController;
+use PKP\core\PKPRequest;
+use PKP\author\contributorRole\ContributorRole;
+use PKP\author\contributorRole\ContributorRoleIdentifier;
+use PKP\author\creditContributorRole\CreditContributorRole;
+use PKP\security\Role;
+use PKP\security\authorization\CanAccessSettingsPolicy;
+use PKP\security\authorization\ContextAccessPolicy;
+use PKP\services\PKPSchemaService;
+
+class ContributorRoleController extends PKPBaseController
+{
+    /**
+     * @copydoc \PKP\core\PKPBaseController::getHandlerPath()
+     */
+    public function getHandlerPath(): string
+    {
+        return 'contributorRoles';
+    }
+
+    /**
+     * @copydoc \PKP\core\PKPBaseController::getRouteGroupMiddleware()
+     */
+    public function getRouteGroupMiddleware(): array
+    {
+        return [
+            'has.user',
+            'has.context',
+            self::roleAuthorizer([
+                Role::ROLE_ID_SITE_ADMIN,
+                Role::ROLE_ID_MANAGER,
+            ]),
+        ];
+    }
+
+    public function authorize(PKPRequest $request, array &$args, array $roleAssignments): bool
+    {
+        $this->addPolicy(new ContextAccessPolicy($request, $roleAssignments));
+        $this->addPolicy(new CanAccessSettingsPolicy());
+
+        return parent::authorize($request, $args, $roleAssignments);
+    }
+
+    /**
+     * @copydoc \PKP\core\PKPBaseController::getGroupRoutes()
+     */
+    public function getGroupRoutes(): void
+    {
+        Route::middleware([
+            self::roleAuthorizer([
+                Role::ROLE_ID_SITE_ADMIN,
+                Role::ROLE_ID_MANAGER,
+            ]),
+        ])->group(function () {
+            Route::get('', $this->getMany(...));
+            Route::get('contributorRoleFormData', $this->getContributorRoleFormData(...));
+            Route::post('', $this->add(...));
+            Route::put('{roleId}', $this->edit(...))
+                ->whereNumber('roleId');
+            Route::delete('{roleId}', $this->delete(...))
+                ->whereNumber('roleId');
```

Comment:
We should also add 
```
Route::get('{roleId}', $this->get(...))
  ->name('contributorRole.getContributorRole')
  ->whereNumber('roleId');
```
 To have consistent API endpoints, and also because we would like `_href` property to be everywhere, also in contributorRole.json.
 
 Also, the routes names are missing, like ` ->name('contributorRole.getContributorRole')` -- s. other API endpoints.